### ISSUE TITLE
Documentation is incorrect for Prometheus troubleshooting

### ISSUE THREAD

**Issue by sshishov** [lensapp/lens#180]
**Describe the bug**
While configuring custom Prometheus according to troubleshooting guide, I was confused which setting should I change in Prometheus configuration

**To Reproduce**
1. Look at README file here: https://github.com/lensapp/lens/blob/936cbd53d400e8973a763373f05514f894a2578a/troubleshooting/custom-prometheus.md

**Expected behavior**
The description should be clear without confusion

**Screenshots**

**Environment (please complete the following information):**
- Lens Version: any
- OS: [e.g. OSX]
- Installation method (e.g. snap or AppImage in Linux): 
 
**Logs:**
When you run the application executable from command line you will see some logging output. Please paste them here:
```
Your logs go here...
```

**Kubeconfig:**
Quite often the problems are caused by malformed kubeconfig which the application tries to load. Please share your kubeconfig, remember to remove any secret and sensitive information.
```
your kubeconfig here
```

**Additional context**

**Comment by alikhil** [comment]
I did everything as it written in readme.md but metrics in the applications still do not work

**Comment by elepedus** [comment]
I too am somewhat confused about the proper way to enable metric in a cluster which already has an existing Prometheus deployment. The documentation appears to pertain to deployments of the Prometheus operator, rather than the Prometheus Helm Chart.

I have a few clusters which were created using GitLab, and Prometheus was installed as a GitLab Managed App, which I believe uses the Helm chart. The metrics work well in GitLab, but unfortunately, I'm not familiar enough with Prometheus to translate the instructions from the readme to work with the Helm chart ConfigMap.

I'm happy to provide a sample of the values.yaml section from the ConfigMap, if it would help improve the readme :)

**Comment by RiFi2k** [comment]
@elepedus I'm trying to get this to work with `gitlab-managed-apps/prometheus-prometheus-server:9090` Prometheus existing install. :grimacing:

I can get the metrics to spin like they are loading now initially which before it was just the metrics not being set up error from the start. Here is what I had to do to get it to where I'm at and maybe someone can figure out what I'm missing.

In the cluster settings...
Prometheus installation method set to `helm`
Prometheus service address set to `gitlab-managed-apps/prometheus-prometheus-server:9090`
This is the namespace the name of the service and the port as installed by GitLab in my cluster.

This is the modified config.yml file adding in the couple things they asked for https://github.com/lensapp/lens/blob/master/troubleshooting/custom-prometheus.md#kube-prometheus

```yml
global:
  evaluation_interval: 1m
  scrape_interval: 30s
  scrape_timeout: 10s
rule_files:
- /etc/config/rules
- /etc/config/alerts
scrape_configs:
- job_name: prometheus
  static_configs:
  - targets:
    - localhost:9090
- bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
  job_name: kubernetes-cadvisor
  kubernetes_sd_configs:
  - role: node
  metric_relabel_configs:
  - regex: (.+)-.+-.+
    source_labels:
    - pod_name
    target_label: environment
  - action: replace
    target_label: instance
  relabel_configs:
  - action: labelmap
    regex: __meta_kubernetes_node_label_(.+)
  - replacement: kubernetes.default.svc:443
    target_label: __address__
  - regex: (.+)
    replacement: /api/v1/nodes/${1}/proxy/metrics/cadvisor
    source_labels:
    - __meta_kubernetes_node_name
    target_label: __metrics_path__
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
  scheme: https
  tls_config:
    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    insecure_skip_verify: true
- job_name: kubernetes-service-endpoints
  kubernetes_sd_configs:
  - role: endpoints
  relabel_configs:
  - action: keep
    regex: true
    source_labels:
    - __meta_kubernetes_service_annotation_prometheus_io_scrape
  - action: replace
    regex: (https?)
    source_labels:
    - __meta_kubernetes_service_annotation_prometheus_io_scheme
    target_label: __scheme__
  - action: replace
    regex: (.+)
    source_labels:
    - __meta_kubernetes_service_annotation_prometheus_io_path
    target_label: __metrics_path__
  - action: replace
    regex: (.+)(?::\d+);(\d+)
    replacement: $1:$2
    source_labels:
    - __address__
    - __meta_kubernetes_service_annotation_prometheus_io_port
    target_label: __address__
  - action: labelmap
    regex: __meta_kubernetes_service_label_(.+)
  - action: replace
    source_labels:
    - __meta_kubernetes_namespace
    target_label: kubernetes_namespace
  - action: replace
    source_labels:
    - __meta_kubernetes_service_name
    target_label: kubernetes_name
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
- bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
  job_name: kubernetes-nodes
  kubernetes_sd_configs:
  - role: node
  metric_relabel_configs:
  - regex: (.+)-.+-.+
    source_labels:
    - pod_name
    target_label: environment
  - action: replace
    target_label: instance
  relabel_configs:
  - action: labelmap
    regex: __meta_kubernetes_node_label_(.+)
  - replacement: kubernetes.default.svc:443
    target_label: __address__
  - regex: (.+)
    replacement: /api/v1/nodes/${1}/proxy/metrics
    source_labels:
    - __meta_kubernetes_node_name
    target_label: __metrics_path__
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
  scheme: https
  tls_config:
    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    insecure_skip_verify: true
- bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
  job_name: kubernetes-pods
  kubernetes_sd_configs:
  - role: pod
  relabel_configs:
  - action: keep
    regex: "true"
    source_labels:
    - __meta_kubernetes_pod_annotation_prometheus_io_scrape
  - action: replace
    regex: (.+)
    source_labels:
    - __meta_kubernetes_pod_annotation_prometheus_io_path
    target_label: __metrics_path__
  - action: replace
    regex: ([^:]+)(?::[0-9]+)?;([0-9]+)
    replacement: $1:$2
    source_labels:
    - __address__
    - __meta_kubernetes_pod_annotation_prometheus_io_port
    target_label: __address__
  - action: labelmap
    regex: __meta_kubernetes_pod_label_(.+)
  - action: replace
    source_labels:
    - __meta_kubernetes_namespace
    target_label: kubernetes_namespace
  - action: replace
    source_labels:
    - __meta_kubernetes_pod_name
    target_label: kubernetes_pod_name
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
  tls_config:
    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    insecure_skip_verify: true
- job_name: autoscaler
  kubernetes_sd_configs:
  - role: pod
  relabel_configs:
  - action: keep
    regex: knative-serving;autoscaler;metrics
    source_labels:
    - __meta_kubernetes_namespace
    - __meta_kubernetes_pod_label_app
    - __meta_kubernetes_pod_container_port_name
  - source_labels:
    - __meta_kubernetes_namespace
    target_label: namespace
  - source_labels:
    - __meta_kubernetes_pod_name
    target_label: pod
  - source_labels:
    - __meta_kubernetes_service_name
    target_label: service
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
  scrape_interval: 3s
  scrape_timeout: 3s
- job_name: activator
  kubernetes_sd_configs:
  - role: pod
  relabel_configs:
  - action: keep
    regex: knative-serving;activator;metrics-port
    source_labels:
    - __meta_kubernetes_namespace
    - __meta_kubernetes_pod_label_app
    - __meta_kubernetes_pod_container_port_name
  - source_labels:
    - __meta_kubernetes_namespace
    target_label: namespace
  - source_labels:
    - __meta_kubernetes_pod_name
    target_label: pod
  - source_labels:
    - __meta_kubernetes_service_name
    target_label: service
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
  scrape_interval: 3s
  scrape_timeout: 3s
- job_name: istio-mesh
  kubernetes_sd_configs:
  - role: endpoints
  relabel_configs:
  - action: keep
    regex: istio-system;istio-telemetry;prometheus
    source_labels:
    - __meta_kubernetes_namespace
    - __meta_kubernetes_service_name
    - __meta_kubernetes_endpoint_port_name
  - source_labels:
    - __meta_kubernetes_namespace
    target_label: namespace
  - source_labels:
    - __meta_kubernetes_pod_name
    target_label: pod
  - source_labels:
    - __meta_kubernetes_service_name
    target_label: service
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
  scrape_interval: 5s
- job_name: istio-policy
  kubernetes_sd_configs:
  - role: endpoints
  relabel_configs:
  - action: keep
    regex: istio-system;istio-policy;http-monitoring
    source_labels:
    - __meta_kubernetes_namespace
    - __meta_kubernetes_service_name
    - __meta_kubernetes_endpoint_port_name
  - source_labels:
    - __meta_kubernetes_namespace
    target_label: namespace
  - source_labels:
    - __meta_kubernetes_pod_name
    target_label: pod
  - source_labels:
    - __meta_kubernetes_service_name
    target_label: service
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
  scrape_interval: 5s
- job_name: istio-telemetry
  kubernetes_sd_configs:
  - role: endpoints
  relabel_configs:
  - action: keep
    regex: istio-system;istio-telemetry;http-monitoring
    source_labels:
    - __meta_kubernetes_namespace
    - __meta_kubernetes_service_name
    - __meta_kubernetes_endpoint_port_name
  - source_labels:
    - __meta_kubernetes_namespace
    target_label: namespace
  - source_labels:
    - __meta_kubernetes_pod_name
    target_label: pod
  - source_labels:
    - __meta_kubernetes_service_name
    target_label: service
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
  scrape_interval: 5s
- job_name: istio-pilot
  kubernetes_sd_configs:
  - role: endpoints
  relabel_configs:
  - action: keep
    regex: istio-system;istio-pilot;http-monitoring
    source_labels:
    - __meta_kubernetes_namespace
    - __meta_kubernetes_service_name
    - __meta_kubernetes_endpoint_port_name
  - source_labels:
    - __meta_kubernetes_namespace
    target_label: namespace
  - source_labels:
    - __meta_kubernetes_pod_name
    target_label: pod
  - source_labels:
    - __meta_kubernetes_service_name
    target_label: service
  - action: replace
    regex: (.*)
    replacement: $1
    source_labels:
    - __meta_kubernetes_pod_node_name
    target_label: kubernetes_node
  scrape_interval: 5s
```

**Comment by wjrogers** [comment]
I have installed Prometheus Operator in the 'monitoring' namespace in my cluster via the [bitnami helm chart](https://github.com/bitnami/charts/tree/master/bitnami/prometheus-operator). I set Lens to "Prometheus Operator" and service address `monitoring/prometheus-operator-prometheus:9090`. I have also followed the [using custom prometheus](https://github.com/lensapp/lens/blob/master/troubleshooting/custom-prometheus.md) instructions and verified that the changes have been applied to the ServiceMonitor specs.

Some metrics appear in Lens. I can see pod CPU, Memory, Network, and Filesystem usage. But the cluster overview says "Metrics not available at the moment," and the Nodes overview resource graphs are blank lines. Despite scouring related issues and the documentation, I can't get this to work at all.

Has anyone gotten this running?

**Comment by emrahcetiner** [comment]
> Some metrics appear in Lens. I can see pod CPU, Memory, Network, and Filesystem usage. But the cluster overview says "Metrics not available at the moment," and the Nodes overview resource graphs are blank lines. Despite scouring related issues and the documentation, I can't get this to work at all.

+1 same issue

**Comment by RiFi2k** [comment]
Took a better look at the docs, just make sure you have installed kube-prometheus as explained in the quickstart or customizing the install explain below it. https://github.com/coreos/kube-prometheus#quickstart

Then depending on what you do, say you have it installed already go find the CRD generated for node-exporter and add the first snippet from the custom prometheus docs into the relabeling, the CRD comes from --> https://github.com/coreos/kube-prometheus/blob/master/jsonnet/kube-prometheus/node-exporter/node-exporter.libsonnet#L176

Then add the second to snippet to the kublet CRD generated from here https://github.com/coreos/kube-prometheus/blob/master/jsonnet/kube-prometheus/prometheus/prometheus.libsonnet#L265

Or use the [jsonnet code they provided](https://github.com/lensapp/lens/blob/master/troubleshooting/custom-prometheus.md#jsonnet) and compile kube-prometheus yourself and it will work out the box.

Also it looks like if you installed the prometheus-operator helm chart you want to add the snippets into `nodeExporter.serviceMonitor.relabelings` and `kubelet.serviceMonitor.metricRelabelings`. Don't quote me but probably like this
```
[{"action": "replace", "regex": "(.*)", "replacement": "$1", "sourceLabels": ["__meta_kubernetes_pod_node_name"], "targetLabel": "kubernetes_node"}]
```
and
```
[{"action": "replace", "sourceLabels": ["node"], "targetLabel": "instance"}]
```

Hopefully that gets anyone else stuck pointed in the right direction.

**Comment by nitrogear** [comment]
@RiFi2k your solution worked for me. I found confusing that https://github.com/lensapp/lens/blob/936cbd53d400e8973a763373f05514f894a2578a/troubleshooting/custom-prometheus.md doesn't clearly describe in what endpoint changes should be made in CRD node-exporter and kubelet.
Here is exactly what helped me:
in CRD node-exporter I had to add new endpoint "https" (I had only "metrics" endpoint):
```
- port: https
  relabelings:
    - action: replace
      regex: (.*)
      replacement: $1
      sourceLabels:
        - __meta_kubernetes_pod_node_name
      targetLabel: instance
```
And in kubelet CRD I added in relabelings of "https-metrics" endpoint:
```
- action: replace
  regex: (.+)
  replacement: $1
  sourceLabels:
    - __metrics_path__
  targetLabel: metrics_path
```

**Comment by mboutet** [comment]
> I have installed Prometheus Operator in the 'monitoring' namespace in my cluster via the [bitnami helm chart](https://github.com/bitnami/charts/tree/master/bitnami/prometheus-operator). I set Lens to "Prometheus Operator" and service address `monitoring/prometheus-operator-prometheus:9090`. I have also followed the [using custom prometheus](https://github.com/lensapp/lens/blob/master/troubleshooting/custom-prometheus.md) instructions and verified that the changes have been applied to the ServiceMonitor specs.
> 
> Some metrics appear in Lens. I can see pod CPU, Memory, Network, and Filesystem usage. But the cluster overview says "Metrics not available at the moment," and the Nodes overview resource graphs are blank lines. Despite scouring related issues and the documentation, I can't get this to work at all.
> 
> Has anyone gotten this running?

Same issue with the bitnami prometheus-operator chart. If I set to "Helm" instead, I can see the overall cluster metrics. However, the resources requests/limits for the pods don't show (they show for the node metrics though). I meticulously followed the instructions in  [custom-prometheus.md](https://github.com/lensapp/lens/blob/master/troubleshooting/custom-prometheus.md).

**Comment by mboutet** [comment]
@wjrogers and @emrahcetiner, I managed to make appear the limits/request for the pods by setting `kube-state-metrics.serviceMonitor.honorLabels` to `true`. I'm also using the bitnami prometheus-operator. However, I need to select "Helm" instead of "Prometheus Operator" for "Prometheus installation method." in Lens.

**Comment by theclue** [comment]
I've followed the solution from @nitrogear and the enanchment from @mboutet, setting as "Prometheus Operator" in Lens settings. It __almost__ works, but some metrics still missing:

- Overall cluster metrics, left panel: still shows "Metrics are not available at the moment"
- Overall cluster metrics, right panel: CPU and Ram Usage are N/A. Pods Usage is ok. Requests and Limits are ok

The same metrics are missing at nodes level, but they works at Pods level.

any idea? Oo'

**Comment by wenerme** [comment]
Here is mine helm values for bitnami/kube-prometheus, rember change installation from auto detect to `Helm` install, prometheus address is `monitoring/kube-prometheus-prometheus:9090`

```yaml
node-exporter:
  serviceMonitor:
    relabelings:
    - action: replace
      regex: (.*)
      replacement: $1
      sourceLabels:
      - __meta_kubernetes_pod_node_name
      targetLabel: kubernetes_node

kubelet:
  serviceMonitor:
    metricRelabelings:
    - action: replace
      sourceLabels:
      - node
      targetLabel: instance

kube-state-metrics:
  serviceMonitor:
    honorLabels: true
```

**Comment by czunker** [comment]
@wenerme settings also work with [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts).

Only the `kube-state-metrics` part is not needed, but set to the same values by the chart.

**Comment by peimanja** [comment]
hmm I could not get it to work with the latest `kube-prometheus-stack` installation.

**Comment by MRostanski** [comment]
@czunker alas, I am not able to reproduce your success - having GKE 1.17.9 and the values for `kube-prometheus-stack` in full are:
```
defaultRules:
  create: false
alertmanager:
  enabled: false
grafana:
  enabled: true
  serviceMonitor:
    enabled: false
  resources:
    limits:
      cpu: 100m
      memory: 128Mi
    requests:
      cpu: 100m
      memory: 128Mi
  persistence:
    type: pvc
    enabled: true
    storageClassName: ssd
    size: 10Gi
  adminPassword: ZAQ12wsx
kubeApiServer:
  enabled: false
kubelet:
  enabled: true
  serviceMonitor:
    metricRelabelings:
      - action: replace
        sourceLabels:
          - node
        targetLabel: instance
kubeControllerManager:
  enabled: false
coreDns:
  enabled: false
kubeEtcd:
  enabled: false
kubeScheduler:
  enabled: false
kubeStateMetrics:
  enabled: true
nodeExporter:
  enabled: true
  serviceMonitor:
    relabelings:
      - action: replace
        regex: (.*)
        replacement: $1
        sourceLabels:
          - __meta_kubernetes_pod_node_name
        targetLabel: kubernetes_node
prometheus:
  enabled: true
  resources:
    limits:
      cpu: 1000m
      memory: 1200Mi
    requests:
      cpu: 1000m
      memory: 1200Mi
prometheusOperator:
  admissionWebhooks:
    enabled: false
```

**Comment by czunker** [comment]
@MRostanski This is our config:
```
---
prometheus:
...

grafana:
...

additionalPrometheusRulesMap:
...

alertmanager:
...

kubelet:
  serviceMonitor:
    https: false
    metricRelabelings:
    - action: replace
      sourceLabels:
      - node
      targetLabel: instance

nodeExporter:
  serviceMonitor:
    relabelings:
      - action: replace
        regex: (.*)
        replacement: $1
        sourceLabels:
        - __meta_kubernetes_pod_node_name
        targetLabel: kubernetes_node
```
Looks like yours for the use of Lens.

But make sure Prometheus is using this config. Take a look at the current config via Web UI: `https://your.host/config`
You should see the relabelings there. Make sure the ServiceMonitor contains the relabelings and perhaps restart the Prometheus Pod.

In case you are using Network Policies, these might also prevent Lens from reaching Prometheus to get the metrics.

**Comment by peimanja** [comment]
In my case closing and reopening Lens fixed my issue after applying the config

**Comment by omninonsense** [comment]
After a lot of head-scratching, it turns out @peimanja was right that it _requires_ a restart for some of the settings to apply (at least the Prometheus one). Reloading (Ctrl+R on Linux) will not work, neither will disconnecting and reconnecting the cluster!

Also, for the people who're trying to get this to work with the GitLab installation of prometheus, you can select Helm as the installation method (that's what Gitlab uses) and put this into the service address field: `gitlab-managed-apps/prometheus-prometheus-server:80`, restart and voila.

**Comment by rafi-fisker** [comment]
Does anyone know how to get the cluster overview metrics working using the [prometheus helm chart ](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus) ?

EDIT: nevermind, somehow it started working after restarting prometheus server

**Comment by tonychoe** [comment]
In my case, the ingress path was the issue.

I set the ingress like this and Lens was not able to discover it automatically.
```
prometheus:
  ...
  ingress:
    enabled: true
    ...
    paths:
      - /prometheus
```
So I have updated the Lens Settings to `Prometheus Operator` and `default/promkube-prometheus:9090/prometheus`, then it worked.

**Comment by Marx2** [comment]
Can anybody provide steps to install custom Prometheus and make it working in lens?
I've tried:
`helm install prometheus prometheus-community/kube-prometheus-stack -n prometheus`
then setting different combination like
`Helm/Prometheus operator prometheus/promkube-prometheus:9090/prometheus`
and nothing works. Lens doesn't see Prometheus at all. Are there any logs in Lens to look at? How to discover correct Prometheus service address?  Which service is the correct one?
<img width="1349" alt="image" src="https://user-images.githubusercontent.com/822886/103456114-79d99a80-4cf3-11eb-81ac-37f933de3335.png">

Edit: I've found correct URL for Lens:
`prometheus/prometheus-kube-prometheus-prometheus:9090/`

**Comment by hobyte** [comment]
> Here is mine helm values for bitnami/kube-prometheus, rember change installation from auto detect to `Helm` install, prometheus address is `monitoring/kube-prometheus-prometheus:9090`
> 
> ```yaml
> node-exporter:
>   serviceMonitor:
>     relabelings:
>     - action: replace
>       regex: (.*)
>       replacement: $1
>       sourceLabels:
>       - __meta_kubernetes_pod_node_name
>       targetLabel: kubernetes_node
> 
> kubelet:
>   serviceMonitor:
>     metricRelabelings:
>     - action: replace
>       sourceLabels:
>       - node
>       targetLabel: instance
> 
> kube-state-metrics:
>   serviceMonitor:
>     honorLabels: true
> ```

@Marx2 use these values and put them in a file (`values.yaml`)
you can then install the chart with `helm install -f values.yaml -n prometheus prometheus bitnami/kube-prometheus`
I tested it with the bitnami chart and it shows the stats in lens

**Comment by viceice** [comment]
@hobyte Does not fully work on kubernetes v1.20.x. Missing the live pod stats. Limits and requests are working.

![image](https://user-images.githubusercontent.com/1798109/107562189-c613d680-6bdf-11eb-9743-207aa5297a46.png)

![image](https://user-images.githubusercontent.com/1798109/107562229-d1ff9880-6bdf-11eb-8560-510fa05980d9.png)

*version*
```
k3s version v1.20.2+k3s1 (1d4adb03)
go version go1.15.5
```

**Comment by viceice** [comment]
looks like the queries used by lens are no longer working on v1.20

lens is using this: `sum(container_memory_working_set_bytes{container!="POD",container!="",pod=~"${opts.pods}",namespace="${opts.namespace}"})`

![image](https://user-images.githubusercontent.com/1798109/107684117-af828380-6ca2-11eb-9b4d-a05a04d19118.png)


if i remove the `container` filter i get values in prometheus: `sum(container_memory_working_set_bytes{pod=~"minio-0",namespace="services"})`

![image](https://user-images.githubusercontent.com/1798109/107684080-a09bd100-6ca2-11eb-9bbc-5f83bb5efd85.png)

**Comment by geerlingguy** [comment]
@RiFi2k - I used the same config for GitLab, but changed the port to 80:

  1. Set installation type to 'Helm'
  2. Set Prometheus address to `gitlab-managed-apps/prometheus-prometheus-server:80`

<img width="800" alt="Screen Shot 2021-02-11 at 12 50 33 PM" src="https://user-images.githubusercontent.com/481677/107684227-23536a80-6c68-11eb-8c87-8f0e705f3a87.png">

And now I'm getting metrics on the Pod and general resource level, but not getting the top-level overview graphs to load. (This is with a cluster where GitLab deployed Prometheus.)

**Comment by WeiTangLau** [comment]
I am able to get the donut chart, but not the line chart at `cluster` page.
The pods charts are working fine

kubernetes version: 1.20.4 (k3s distro)
lens version: 4.1.3

![image](https://user-images.githubusercontent.com/42403590/109748772-645def80-7c14-11eb-8b8a-d70963f35b87.png)

![image](https://user-images.githubusercontent.com/42403590/109748841-85264500-7c14-11eb-83f1-959e801cf022.png)

**Comment by viceice** [comment]
Just to report back, it's now working again after k3s update 1.20.4-k3s1 (i think k3s-io/k3s#2831 was the root cause)

Now using `kube-prometheus` from bitnami with this values:
```yaml
node-exporter:
  serviceMonitor:
    relabelings:
    - action: replace
      regex: (.*)
      replacement: $1
      sourceLabels:
      - __meta_kubernetes_pod_node_name
      targetLabel: kubernetes_node

kubelet:
  serviceMonitor:
    metricRelabelings:
    - action: replace
      sourceLabels:
      - node
      targetLabel: instance

kube-state-metrics:
  serviceMonitor:
    honorLabels: true

# no docker hub limits 😅 
operator:
  image:
    registry: gcr.io
    repository: bitnami-containers/prometheus-operator

prometheus:
  image:
    registry: gcr.io
    repository: bitnami-containers/prometheus

alertmanager:
  image:
    registry: gcr.io
    repository: bitnami-containers/alertmanager
```

lens detection is left on automatic

**Comment by z0rc** [comment]
This whole discussion raises a question: Why metrics relabeling is needed in the first place? Did Lens pick some arbitrary labels and tries to force them on various configurations?

It's okay to relabel metrics when you're just testing things out. But once you have production environment with dashboards and other services query prometheus (like prometheus-adapter) relabeling is a breaking change.

Have developers considered approach where user is allowed to provide needed queries and/or labels in Lens configuration?

**Comment by jakolehm** [comment]
> This whole discussion raises a question: Why metrics relabeling is needed in the first place? Did Lens pick some arbitrary labels and tries to force them on various configurations?

Good question! Basically it's needed because many of the prometheus installation methods use different labels. It would have been much easier for Lens if there would be some kind of standard. (Lens tries to auto-detect used prometheus configuration but it's hard to keep up with all the prometheus charts/configs)

**Comment by xvonabur** [comment]
@omninonsense, I've tried to use Gitlab Prometheus, as you mentioned above, but CPU/Memory limits and usage are N/A. Lens displays CPU and Memory requests only. Did you do some relabeling?

**Comment by s1rc** [comment]
@xvonabur @omninonsense I am also wondering the same. I'm attempting install Prometheus using the instructions here: https://docs.gitlab.com/ee/user/clusters/integrations.html#prometheus-prerequisites

I am setting `scrape_interval` at `30s`, I can see some metrics at the pod level but not top level cluster/node metrics. Perhaps the `values.yml` provided by GitLab in the instructions require additional relabeling.

**Comment by VitaliAndrushkevich** [comment]
Collegues, if you have troubles with visibility Cluster and Node info after installing **kube-prometheus-stack**, check availability of kube-state-metrics (I mean ServiceMonitors), especially you should have available target in `Prometheus->Status->Targets->*kube-state-metrics/0 (1/1 up)`.

In my case I've changed port name to `http`, since by default in Servicemonitor it's `metrics`, but in the service of **kube-state-metrics** by default is `http`:
```yaml
kind: ServiceMonitor
apiVersion: monitoring.coreos.com/v1
...
spec:
  endpoints:
    - honorLabels: true
      port: http
  jobLabel: app.kubernetes.io/name
  selector:
    matchLabels:
      app.kubernetes.io/instance: prometheus
      app.kubernetes.io/name: kube-state-metrics
```

**Comment by errnesto** [comment]
same here with gitlab-managed apps / prometheus-community/prometheus
setting the url to `gitlab-managed-apps/prometheus-prometheus-server:80` gives me metrics on pods (except for 'Filesystem') but not for the nodes or cluster overview…

**Comment by marcbachmann** [comment]
Fixes coming in #3653, using the default [`kube-prometheus-stack`](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) setup
